akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -224,7 +225,7 @@ import scala.collection.JavaConverters._
     parent ⇒
     lazy val decider = inheritedAttributes.mandatoryAttribute[SupervisionStrategy].decider
     private val activeSubstreamsMap = new java.util.HashMap[Any, SubstreamSource]()
-    private val closedSubstreams = new java.util.HashSet[Any]()
+    private val closedSubstreams = if (allowClosedSubstreamRecreation) Collections.unmodifiableSet(new java.util.HashSet[Any]) else new java.util.HashSet[Any]()
```

Comment:
`Collections.unmodifiableSet(Collections.emptySet) ` this is the case where we never want to touch this collection -- this also means we'll never "infinitely accumulate keys"

Reply:
Is this about changing documentation or "just" change the code to your suggestion?

Reply:
Actually just to `Collections.emptySet` since it is immutable already :) yes just code